packages/eslint-plugin/tests/rules/no-unnecessary-type-assertion.test.ts

```diff
@@ -117,6 +124,19 @@ function testFunction(_param: string | null): void { /* noop */ }
 const value = 'test' as string | null | undefined
 testFunction(value!)
     `,
+    // https://github.com/typescript-eslint/typescript-eslint/issues/982
+    {
+      code: `
+const React = require("react");
+
+function Test(props: {
+  id?: null | string | number;
+}) {
+  return <div key={props.id!} />;
+};
+      `,
+      filename: path.join(rootDir, 'react.tsx'),
```

Comment:
filename for now has to be absolute: #1424 should fix this

Reply:
do we want to put this in the `RuleTester`?

Reply:
no it should not be there, it should work correctly with relative paths
see #1439